### ISSUE TITLE
NER for fractions

### ISSUE THREAD

**Issue by jonzim** [winkjs/wink-eng-lite-model#6]
the entity name patterns used to recognize numeric entities appear to be lacking - i can't recognize fractions... and decimals also don't seem to pick up correctly.   For example when used to extract the parts of a sentence indicating ingredients i'm using this custom pattern:

    const patterns = [
      { name: 'number', patterns: [ '[|PERCENT] [|CARDINAL] [|ORDINAL]' ] },
      { name: 'measurement', patterns: [ 'cup' ] },
      { name: 'food', patterns: [ 'butter' ] },
      { name: 'adjective', patterns: [ 'ADJ' ] },
      { name: 'adverbPhrase', patterns: [ '[|ADV] [|VERB]' ] }
    ];

then i try this string

"1/2 cup yellow butter, softened."  ---> 1/2 was not picked up: 
```
  [Object: null prototype] { value: 'cup', type: 'measurement' },
  [Object: null prototype] { value: 'yellow', type: 'adjective' },
  [Object: null prototype] { value: 'butter', type: 'food' },
  [Object: null prototype] { value: 'softened', type: 'adverbPhrase' }
```

".5 cup yellow butter, softened." ---> .5 was changed to 5
`   [Object: null prototype] { value: '5', type: 'number' },`

but 0.5 worked, and "one half" also worked.

I expected NUM to be defined, but apparently you are not using that one.  https://universaldependencies.org/u/pos/all.html#al-u-pos/NUM

**Comment by sanjayaksaxena** [comment]
Hello @jonzim 

Many thanks for highlighting the issue. 

The **NUM** is supported and it works. The fractions like **1/2** are not getting detected due to a text pre-processing miss — we will fix it shortly — until such time we request you to use **NUM**. As of now **.5** as a number is not supported and you may create a custom entity to detect the same.

Following code should help you:
```javascript
import winkNLP from 'wink-nlp';
import model from 'wink-eng-lite-web-model';


const nlp = winkNLP( model );
const its = nlp.its;

const text = 'I would just like to eat 1/2 or 0.5 or one half cup soft butter.';
const patterns = [
  { name: 'number', patterns: [ '[|PERCENT|NUM|ORDINAL|CARDINAL]' ] },
  { name: 'measurement', patterns: [ 'cup' ] },
  { name: 'food', patterns: [ 'butter' ] },
  { name: 'adjective', patterns: [ 'ADJ' ] },
  { name: 'verbPhrase', patterns: [ '[|ADV] [VERB]' ] },
  { name: 'verbPhrase', patterns: [ '[|ADV] [VERB] [to] [VERB]' ] }
];
nlp.learnCustomEntities( patterns, {
  matchValue: false,
  usePOS: true,
  useEntity: true
} );

const d = nlp.readDoc(text);

d.customEntities().each( e => e.markup( '{', `}/${e.out(its.type)}`) );

console.log( d.out(its.markedUpText) )
// Outputs:
// -> I would {just like to eat}/verbPhrase {1/2}/number or {0.5}/number or {one half}/number {cup}/measurement {soft}/adjective {butter}/food.
```
<sup>**Tested on Node.js version 16.13.0**</sup>

Best,
Sanjaya

**Comment by jonzim** [comment]
Perfect, that is working for me.

thank you.  I did not realize NUM was programmed in as it was not documented here - I assumed it was the full list of supported values - my bad:  https://github.com/winkjs/wink-eng-lite-model. 

thank you very much for the tips!

**Comment by sanjayaksaxena** [comment]
fixing it in web model first — refer https://github.com/winkjs/wink-eng-lite-web-model/commit/61be004ba4346117ff9395bc4d1b991aa05578a3

**Comment by sanjayaksaxena** [comment]
Published the latest version (1.4.2) of [wink-eng-lite-web-model](https://github.com/winkjs/wink-eng-lite-web-model) on [npm](https://www.npmjs.com/package/wink-eng-lite-web-model).
